### PR TITLE
fix histograms

### PR DESCRIPTION
**Pull request by harabat**
Implements solution discussed in issue #2411 (based on https://github.com/vega/vega-lite/issues/5403) for the examples [Layered Histogram](https://altair-viz.github.io/gallery/layered_histogram.html) and [Facetted Scatterplot with marginal histograms](https://altair-viz.github.io/gallery/scatter_marginal_hist.html).

### CODE REVIEW

**Comment by joelostblom** [comment]
I built locally and it looks good to me, thanks! You can see that the binned histograms extends a bit longer than the area one, which I believe is the most correct approach so that the last bin is not half width as previously:

image

image

**Review by joelostblom — APPROVED** [review]

**Comment by harabat** [comment]
Hadn't noticed that half-bin, good catch.

File: altair/examples/scatter_marginal_hist.py [hunk 1]
```diff
@@ -16,15 +16,15 @@
 xscale = alt.Scale(domain=(4.0, 8.0))
 yscale = alt.Scale(domain=(1.9, 4.55))
 
-area_args = {'opacity': .3, 'interpolate': 'step'}
+area_args = {'opacity': .3, 'binSpacing': 0}
```
Comment (jakevdp) [inline altair/examples/scatter_marginal_hist.py:19]: nitpick: change this to `bar_args`?

Comment (harabat) [reply altair/examples/scatter_marginal_hist.py:19]: Good catch, I'll change this.

Comment (harabat) [reply altair/examples/scatter_marginal_hist.py]: @jakevdp done

**Pull request merged**